File: ege/26solve/26-107.cpp
// Автор: М. Нехорошева

#include<iostream>
#include<vector>
#include<algorithm>
#include<fstream>
using namespace std;
int main()
{
	fstream F;
	F.open("26-107.txt");
	vector<pair<int,int> >a;
	int n,L;
	F>>L>>n;
	for (int x,y;F>>x>>y;a.emplace_back(y,x));
	sort(a.begin(),a.end());
	int k=1,last=a[0].first,time_last;
	for (int i=1;i<a.size();++i)
		if (a[i].second>=last)
			++k,last=a[i].first,time_last=a[i].second;
	cout<<k<<"  "<<	time_last;
}

File: ege/26solve/26-107.py
# Автор: Д. Статный

f = open('26-107.txt')
l, n = map(int, f.readline().split())
a = [list(map(int, f.readline().split())) for _ in range(n)]
a.sort(key = lambda x: (x[1], x[0]))
flights = [[a[0][0], a[0][1]]]
for x in a[1:]:
    if x[0] >= flights[-1][1]:
        flights.append([x[0], x[1]])
print(len(flights), flights[-1][0])
